### DISCUSSION TITLE
How can I make javadoc of custom component available in scripts

### DISCUSSION THREAD

**Discussion by JensVercruysse** [nextgenhealthcare/connect#5967] [Q&A] [answered]
I have imported my custom jar that has a javadoc included in the custom-lib, is there any way to make this javadoc available in deploy and transformer scripts?
Just like it's available for built in libs like DateUtil. Do I have to include something in my jar to make this work? There is a user-api in my jar already with the full javadoc.

**Comment by jonbartels** [comment, marked as answer]
https://github.com/nextgenhealthcare/connect/blob/010f2a4dc636ad1c24b20a7fd1f625c5f6b91a39/client/src/com/mirth/connect/client/ui/reference/ReferenceListFactory.java#L538

That code populates things like the popups you get for `DateUtil` and `FileUtil`. It does appear to load the data from extensions.

The extension has to define `userUtilPackages` like this https://github.com/nextgenhealthcare/connect/blob/010f2a4dc636ad1c24b20a7fd1f625c5f6b91a39/server/src/com/mirth/connect/plugins/httpauth/plugin.xml#L14

You might also be able to have your classes in the package `com.mirth.connect.userutil` since that loads by scanning the classpath. I don't like that solution because your classes aren't part of that package. 

I know I've seen this question before but I cannot find a reference.

**Reply by JensVercruysse** [reply]
Thank you, I will take a look and see if I can get it working
